crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -1,680 +1,104 @@
-}
-
-/// A helper function to keep the backoff strategy construction separated.
-async fn retry<T, E, Fut, FutureFactory, RetryCondition>(
-    future_factory: FutureFactory,
-    retry_condition: RetryCondition,
-) -> Result<T, E>
-where
-    Fut: Future<Output = Result<T, E>>,
-    FutureFactory: FnMut() -> Fut,
-    RetryCondition: FnMut(&E) -> bool,
-{
-    Retry::exponential(future_factory, NonZeroU64::new(2).unwrap())
-        .factor(NonZeroU64::new(15).unwrap())
-        .max_delay(Duration::from_secs(10 * 60))
-        .when(retry_condition)
-        .await
-}
-
-#[derive(Clone)]
-struct EthereumImpl<T: EthereumTransport + Send + Sync> {
-    logs: Arc<RwLock<StateRootFetcher>>,
-    transport: T,
-}
+use pathfinder_ethereum::{L1StateUpdate, StarknetEthereumClient};
+use tokio::sync::mpsc::Sender;
+
+/// Syncs L1 state updates.
+pub async fn sync(
+    tx_event: Sender<L1StateUpdate>,
+    ethereum_client: StarknetEthereumClient,
+    poll_interval: std::time::Duration,
+) {
+    let mut backoff = RetryBackoff::new(std::time::Duration::from_secs(1), poll_interval);
+    loop {
+        tokio::time::sleep(backoff.delay()).await;
 
-#[async_trait::async_trait]
-impl<T: EthereumTransport + Send + Sync + Clone> EthereumApi for EthereumImpl<T> {
-    async fn fetch_logs(&mut self) -> Result<Vec<StateUpdateLog>, FetchError> {
-        let ff = || async {
-            let logs = self.logs.clone();
-            let transport = self.transport.clone();
-            let mut logs = logs.write().await;
-            logs.fetch(transport).await
-        };
-        let logs = retry(ff, |error| match error {
-            FetchError::Other(other) => {
-                tracing::warn!(reason=%other, "Failed fetching L1 logs, retrying");
-                true
+        match ethereum_client.get_starknet_state().await {
+            Ok(state) => {
+                backoff.success();
+                if let Err(e) = tx_event.send(state).await {
+                    tracing::error!(reason=?e, "L1 event sending failed");
+                }
             }
-            FetchError::Reorg => false,
-        })
-        .await?;
-        Ok(logs)
-    }
-
-    async fn set_log_head(&mut self, head: Option<StateUpdateLog>) {
-        self.logs.write().await.set_head(head);
-    }
-
-    async fn log_head(&self) -> Option<StateUpdateLog> {
-        self.logs.read().await.head().clone()
-    }
-
-    async fn block_hash(
-        &self,
-        block: EthereumBlockNumber,
-    ) -> anyhow::Result<Option<EthereumBlockHash>> {
-        // No explicit retrying here as any `EthereumTransport` implementor should already hadle that there.
-        Ok(self
-            .transport
-            .block(block.0.into())
-            .await?
-            .map(|b| EthereumBlockHash(b.hash.unwrap().0.into())))
+            Err(e) => {
+                backoff.failure();
+                tracing::error!(reason=?e, "L1 call failed");
```

Comment:
We should probably lower this -- unless we have reason to believe these should never fail? Network failures are common and not unexpected, `Error` has users freak out -- these should be reserved for really major things, or things they should report to us.
```suggestion
                tracing::debug!(reason=?e, "L1 call failed");
```